week-6/Homework/mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,39 @@
-function setAlarm() {}
+function setAlarm() {
+  let count = document.getElementById("timeRemaining");
+  let setIt = document.getElementById("set");
+  let pause = document.getElementById("p");
+  let Restart = document.getElementById("re");
+  let Counter;
+  pause.addEventListener("click", function () {
+    clearInterval(Counter);
+  });
+  Restart.addEventListener("click", function () {
+    Counter = setInterval(CountDown, 1000);
+  });
+  let TheTime;
+  setIt.addEventListener("click", function () {
+    TheTime = document.getElementById("alarmSet").value;
```

Comment:
I would rather parse this to a number because even if the type of the input box is number this function will still return a string. JS is quite lenient so let you threat string as a number in given circumstances but it is not a good practice and can lead to a cunning bug later on.